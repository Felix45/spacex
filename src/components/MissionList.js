import React from 'react';
import PropTypes from 'prop-types';
import Table from 'react-bootstrap/Table';
import Mission from './Mission';

const MissionList = (props) => {
  const { missions } = props;

  return (
    <Table striped bordered responsive>
      <thead>
        <tr>
          <th width="100">Mission</th>
          <th>Description</th>
          <th width="100">Status</th>
          <th width="150"> </th>
        </tr>
      </thead>
      <tbody>
        {
          missions.map((mission) => (
            <Mission key={mission.id} mission={mission} />))
        }
      </tbody>
    </Table>
  );
};

MissionList.propTypes = {
  missions: PropTypes.instanceOf(Object).isRequired,
};

export default MissionList;
