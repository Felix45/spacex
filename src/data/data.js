const ships = [
  {
    id: 1,
    rocket_name: 'Falcon1',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://i.imgur.com/DaCfMsj.jpg',
  },
  {
    id: 2,
    rocket_name: 'Falcon9',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
  },
  {
    id: 3,
    rocket_name: 'Starship',
    description: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!',
    flickr_images: 'https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg',
  },
];

export default ships;
