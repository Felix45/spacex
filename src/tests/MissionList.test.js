import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux';
import { missionData } from '../data/mock';
import MissionList from '../components/MissionList';

test('MissionsList component renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MissionList missions={missionData} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
