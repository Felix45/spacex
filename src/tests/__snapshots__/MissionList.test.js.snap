// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsList component renders correctly 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr>
        <th
          width="100"
        >
          Mission
        </th>
        <th>
          Description
        </th>
        <th
          width="100"
        >
          Status
        </th>
        <th
          width="150"
        >
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="fw-bold"
        >
          Thaicom
        </td>
        <td>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
        </td>
        <td
          className="align-middle"
        >
          <span
            className="text-uppercase badge bg-secondary"
          >
            Not a member
          </span>
        </td>
        <td
          className="align-middle"
        >
          <button
            className="w-100 text-uppercase btn btn-outline-dark"
            disabled={false}
            id={1}
            onClick={[Function]}
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
      <tr>
        <td
          className="fw-bold"
        >
          Telstar
        </td>
        <td>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
        </td>
        <td
          className="align-middle"
        >
          <span
            className="text-uppercase badge bg-info"
          >
             Active Member 
          </span>
        </td>
        <td
          className="align-middle"
        >
          <button
            className="w-100 text-uppercase btn btn-outline-danger"
            disabled={false}
            id={2}
            onClick={[Function]}
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
      <tr>
        <td
          className="fw-bold"
        >
          JCSAT
        </td>
        <td>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsam, consequatur ut. Distinctio nostrum voluptatibus recusandae eum, velit dolores mollitia nam aliquid consectetur enim corporis perspiciatis repudiandae animi, sed hic tenetur!
        </td>
        <td
          className="align-middle"
        >
          <span
            className="text-uppercase badge bg-info"
          >
             Active Member 
          </span>
        </td>
        <td
          className="align-middle"
        >
          <button
            className="w-100 text-uppercase btn btn-outline-danger"
            disabled={false}
            id={3}
            onClick={[Function]}
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
